pkg/test/echo/response/checkcall/checkers.go

```diff
@@ -0,0 +1,123 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package checkcall
+
+import (
+	"errors"
+	"fmt"
+
+	"istio.io/istio/pkg/test/echo/response"
+	"istio.io/istio/pkg/test/framework/components/cluster"
+)
+
+// And is an aggregate Checker that requires all Checkers succeed.
+func And(checkers ...Checker) Checker {
+	return Func(func(r response.Instances, err error) error {
+		for _, c := range checkers {
+			if err := c.CheckCall(r, err); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+func Or(checkers ...Checker) Checker {
+	return Func(func(r response.Instances, err error) error {
+		var lasterr error
+		for _, c := range checkers {
+			if e := c.CheckCall(r, err); e != nil {
+				lasterr = err
+				continue
+			}
+			return nil
+		}
+		return fmt.Errorf("all checks failed. Last error: %v", lasterr)
```

Comment:
nit: says "all" failed but could be only one..?

depending on usage it might look weird too. if I say `or(code(200), code(201), code(202))`, I would get "all checks failed. Last error: expected code 202, got code 300".

Reply:
I think `Or` is confusing altogether. I've removed it and modified the only call site that uses it.